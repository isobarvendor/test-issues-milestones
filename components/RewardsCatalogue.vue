<template>
  <div class="rewards-catalogue">
    <div class="wallet-desktop" >
      <div class="wallet-swiper" v-for="(item, index) in data.prizeList" :key="'reward'+index"  @click="goDetail(index)">
        <div class="wallet-swiper-item">
          <img :src="item.imgUrl"
          />
        </div>
        <div class="description bg">
            <div class="name">{{item.name}}</div>
            <div class="expiry" :key="'cost'+index">
              <span>{{item.amountAvailable}}</span>
              <span>Coins</span>
            </div>
          </div>
      </div>
    </div>
  </div>
</template>

<script>
export default {
  name: "RewardsCatalogue",
  components: {
  },
  data() {
    return {
      currentPage: 1
    };
  },
  props: {
    data: null,
  },
  computed:{
  	toBeShown() {
    	return this.data.exclusivePrizes.slice(0, this.currentPage * 3);
    },
    totalPages() {
    	return Math.ceil( this.data.exclusivePrizes.length / 3);
    }
  },
  methods:{
    nextPage(){
    	if(this.currentPage <  this.totalPages) this.currentPage++;
    },
    prevPage(){
    	this.currentPage = this.currentPage - 1 || 1;
    },
    goDetail(rewardID){
      this.$router.push({ path: 'rewardsDetail/'+rewardID})
    }
  },
  beforeMount() {},
  mounted(){

  }
};
</script>

<style lang="scss">
.rewards-catalogue{
    margin-bottom: 40px;
    .wallet-desktop{
      display: flex;
      flex-wrap: wrap;
      .wallet-swiper{
        margin: 30px 20px;
        width: 29%;
      }
      .description{
        padding: 20px;
        border-radius: 0px 0px 20px 20px;

        .name:after{
          content: '';
          display: block;
          margin: .5em 0;
          height: 1px;
          background: #585858;

        }
        .expiry{
          display: flex;
          justify-content: flex-end;
          span{
            font-size:12px;
          }
          span:first-child{
            margin-right: 0.5em;
            font-weight: bold;
            font-size: 26px;
          }
        }
      }
    }
    .prize-swiper-pagination > * + *:before{
      background: #585858
    }
    .prize-swiper-pagination .prize-swiper-pagination-bullet{
      background: #585858
    }
    .prize-swiper-pagination .prize-swiper-pagination-bullet-active{
      background: #ffffff
    }
    @media only screen and (max-width: 1199px) {
    .wallet-desktop{
      .wallet-swiper{
          width: 43%;
          margin: 30px 10px;
        .description{
          padding: 1em 10px;
          .name{
            font-size: 14px;
          }
          .expiry{
            justify-content: flex-start;
            span:first-child{
              font-size: 14px;
            }
          }
        }

      }
    }
    }

}
</style>
