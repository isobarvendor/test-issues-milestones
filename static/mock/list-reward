[ {       "public_code": "shopee-test-2",
       "secret_code": "005ZWWZXFE82",
       "url": "https://barcode-stage.web.app/c/shopee-test-2",
        "secret_code_image": "/img/landing/barcode.png",
       "store_name": "Shopee",
       "coupon_expired_at": "2021-05-09 17:50:00",
       "creative_image": "https://cdn2.grivy.com/Testing%20Juice/JUICE_BANNER-1080x480.jpg"
   },
   {       "public_code": "shopee-test-2",
       "secret_code": "005ZWWZXFE82",
       "url": "https://barcode-stage.web.app/c/shopee-test-2",
        "secret_code_image": "/img/landing/barcode.png",
       "store_name": "Shopee",
       "coupon_expired_at": "2021-05-09 17:50:00",
       "creative_image": "https://cdn2.grivy.com/Testing%20Juice/JUICE_BANNER-1080x480.jpg"
   },
   {       "public_code": "shopee-test-2",
       "secret_code": "005ZWWZXFE82",
       "url": "https://barcode-stage.web.app/c/shopee-test-2",
        "secret_code_image": "/img/landing/barcode.png",
       "store_name": "Shopee",
       "coupon_expired_at": "2021-05-09 17:50:00",
       "creative_image": "https://cdn2.grivy.com/Testing%20Juice/JUICE_BANNER-1080x480.jpg"
   }
]
