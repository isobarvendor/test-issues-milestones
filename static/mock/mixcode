{
                      "programId": 453269,
                      "code": "3WTD90GE5",
                      "lot": {
                        "lotId": 4455843436,
                        "lotName": "Coke 500ml PET",
                        "activateDate": 1611208800000,
                        "inactivateDate": 1627603200000,
                        "active": true,
                        "expired": false,
                        "bevProdPkg": {
                          "trademarkCd": "05",
                          "trademarkName": "Coca-Cola",
                          "brandCd": "001",
                          "brandName": "Coca-Cola",
                          "bevProdCd": "0001",
                          "bevProdName": "Coca-Cola",
                          "caffeinated": true,
                          "carbonated": true,
                          "calorieCategory": {
                            "code": 1,
                            "description": "Regular"
                          }
                        },
                        "releaseId": 494494,
                        "releaseName": "Coke PET",
                        "organizationId": 157168,
                        "organizationName": "Harish",
                        "realtimeCode": false,
                        "testRelease": false,
                        "pointValue": 0,
                        "bevProdOptions": []
                      },
                      "redeemed": false,
                      "actualCode": "3WTD90GE5"
                    }
