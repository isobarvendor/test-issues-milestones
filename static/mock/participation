{"id":16,"questionId":1,"answerId":3,"participationId":17}
