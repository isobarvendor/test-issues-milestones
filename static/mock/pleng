{
  "burnResult": [
    {
      "pincode": "0J6TC2VP5",
      "programId": "453269",
      "lotId": "445580",
      "burned": true
    }
  ],
  "participationId": "1ylazdy9kkxmpa53",
  "instantWinResult": {
    "winner": true,
    "redeemedPrize": {
      "barcodeType": 1,
      "deliveryType": 1,
      "active": true,
      "description": "<p>3 months subscription to Pleng Pro</p>",
      "shortDescription": "<p>ឈ្នះថ្លៃសេវា Pleng Pro Subscription រយៈពេល 3 ខែ</p>",
      "priority": 1,
      "prizeId": "ks8o4ixh",
      "imgUrl": "https://d1tnwdlrciourl.cloudfront.net/prizeImages/z2xtqi9ks8ndwx3/ks8o4ixh/CAM prizes — Pleng Pro@2x.png",
      "name": "3 months subscription to Pleng Pro ",
      "redemptionLink": "https://pleng.onelink.me/pp89?pid=ddlink&af_web_dp=https%3A%2F%2Fwww.smart.com.kh%2Fkm%2Fpleng%2F&af_dp=plengmusic%3A%2F% 2F&af_partner_name=cc",
      "configurationId": "z2xtqi9ks8ndwx3",
      "redeemDescription": "<p>អបអរសាទរ! អ្នកបានឈ្នះ Pleng Pro Subscription រយៈពេល ៣ ខែ! រីករាយនូវការស្តាប់ចម្រៀងឥតដែនកំណត់។ សូមដោនឡូតកម្មវិធី Pleng ពី Smart ដើម្បីបញ្ចូលលេខកូដ និងដាក់ដំណើរការ Smart Pleng Pro។ សូមប្តូរយករង្វាន់ឲ្យបានមុនថ្ងៃទី ៣០ ខែ វិច្ឆិកា ឆ្នាំ ២០២១។</p>",
      "voucherCode": "CNXFUUAKAY"
      
    }
  }
}