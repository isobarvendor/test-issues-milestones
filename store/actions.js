
import GeneralAPI from '../api/general';

import {LOGIN, SIGNUP, GET_ACCOUNT} from './action_types';

export default {
  [LOGIN]: ({ commit, state, getters }, data) => {
    return new Promise((resolve, reject) => {
      const moduleState = state;
      if (moduleState.login) {
        return resolve(moduleState.login)
      } else {
        GeneralAPI.login(data)
        .then(response => {
          commit('SET_TOKEN', response.data.token);
          sessionStorage.token=response.data.token;
          return resolve(messages);
        })
        .catch(error => {

          return reject(error);
        });
      }
    });
  },
  [GET_ACCOUNT]: ({ commit, state, getters }, token) => {
    return new Promise((resolve, reject) => {
      const moduleState = state;
      if (moduleState.login) {
        return resolve(moduleState.login)
      } else {
        GeneralAPI.getAccount(token)
        .then(response => {

          commit('SET_LOGIN_ACCOUNT', response.data);
          return resolve(messages);
        })
        .catch(error => {

          return reject(error);
        });
      }
    });
  },
  [SIGNUP]: ({ commit, state, getters }, data) => {
    return new Promise((resolve, reject) => {
      const moduleState = state;
      if (moduleState.login) {
        return resolve(moduleState.login)
      } else {
        GeneralAPI.signup(data)
        .then(messages => {
          //commit('SET_LOGIN_ACCOUNT', messages);
          return resolve(messages);
        })
        .catch(error => {
          console.error(error);
          return reject(error);
        });
      }
    });
  },


  /*[SET_GLOBAL_ERROR_DIALOG]:({commit, state}, status)=>{
    commit('SET_GLOBAL_ERROR_DIALOG', status);
  }
  */
};
