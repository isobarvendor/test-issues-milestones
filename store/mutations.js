

//shared store are concatenated to the central store
//TO-DO: manage the shared store seperately by module
export default {

  SET_LOGIN_ACCOUNT:(state, data)=>{
        state.login = data;
    },

  SET_TOKEN:(state, data)=>{
      state.token = data;
  },
  SET_ERROR:(state, data)=>{
    state.errorLogin = data;
},
  SET_LIST_PRIZE:(state, data)=>{
    state.listPrize = data;
  },
  SET_LIST_WALLET:(state, data)=>{
    state.listWallet = data;
  },
  SET_FILE_AMAZON:(state, data)=>{
    state.fileUploaded = data;
  },
  SET_CONFIG:(state,data)=>{
    state.config = data
  },
  SET_CMS_CONTENT:(state,data)=>{
    state.CMSContent = data
  }



};
