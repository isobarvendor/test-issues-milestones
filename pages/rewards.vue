<template>
  <div v-if="dataStatus.status == 200" id="main" class="rewards">
    <div class="wrapper">

      <div class="container">
        <div class="header"> <span class="title">Rewards Catalogue</span>
          <div class="selections">
           <!-- <v-select
              :items="recent"
              label="Most recent"
              item-text ="text"
              item-value = "value"
              class="select"
            ></v-select>
            <v-select
              :items="filterby"
              label="Filter by"
              item-text ="text"
              item-value = "value"
              class="select"
            ></v-select>
            <v-text-field
              label="Search"
              solo
              prepend-inner-icon="mdi-magnify"
              v-model="search"
              class="search"
            ></v-text-field> -->
          </div>
        </div>
        <RewardsCatalogue :data="listPrizes"/>
      </div>

    </div>
  </div>
  <div v-else-if="dataStatus.status >= 500">
    Status: {{dataStatus.status}}
    <br />
    {{dataStatus.message}}
  </div>
</template>

<script>
import RewardsCatalogue from '@/components/RewardsCatalogue';
import { GET_LIST_PRIZE } from '@/store/action_types';
import deepClone from 'deep-clone'
// import Vuetify from 'vuetify/lib';
export default {
  data(){
    return{
      dataStatus:{},
      data:null,
      configData: null,
      campaignType: 0,
      listPrizes :[],
      listPrizesData :[],
      search:null
    }
  },
  head() {
    return {
      title: "Coke Campaign Title",
      meta: [
        {
          hid: "description",
          name: "description",
          content: "Home page description"
        }
      ],
      noscript: [],
      script: [
        {
          hid: "anime",
          src: "/js/anime.min.js"
        }
      ],
      css: []
    };
  },
  computed: {
    recent(){
        return this.data  && this.data.exclusivePrizes
    },
    filterby(){
        return this.data && this.data.exclusivePrizes
    }
  },

  created() {
    this.fetchData();
    this.getListPrize();
  },
  methods: {
    async fetchData() {

      let result = await this.$axios.get(
        "https://ayo.aircovery.com/cms-api/campaigns"
      );
      this.dataStatus = { status: result.status, message: result.statusText };
      this.data = result.data;

      let config = await this.$axios.get('https://ayo.aircovery.com/cms-api/campaign-configurations')
      this.configData = config.data

    },
    async getListPrize(){


          await  this.$store.dispatch(GET_LIST_PRIZE)
            .then((response)=>{
               this.listPrizes=deepClone(response.data);
               this.listPrizesData=deepClone(response.data);
            })
            .catch((error) =>{
              if(error.response && error.response.data.status=='401'){
                this.errorMessage='Please enter the correct email/password';
              }
            })


    },

  },
  watch:{
    search: function(val){
      if(val&&val!=''){
        this.listPrizes.prizeList=this.listPrizes.prizeList.filter((o)=>{
            let searchVal=val.toUpperCase();

            return o.name.toUpperCase().includes(searchVal) ||o.shortDescription.toUpperCase().includes(searchVal)||o.description.toUpperCase().includes(searchVal);
        })
      }
      else{

        this.listPrizes=deepClone(this.listPrizesData);
      }
    }
  }
};
</script>

<style lang="scss">
@import '~assets/scss/variables.scss';
@import '~assets/scss/mixins.scss';
  html, body{
    background: $body-bg-color-wallet;
    .rewards{
        padding-top:115px;

        .container{
            max-width:1280px;
            padding: $content-hpadding $content-hpadding;
            @include breakpoint(lg){
            padding: $masthead-vpadding $masthead-hpadding;
            }
        }

        .header{
            @include section-header;
            text-align:left;
            text-transform: capitalize;
            display: flex;
            font-size:28px;
            justify-content: space-between;
            align-items: center;
            .selections{
                display: flex;
                .v-input{
                    margin-left: 30px;
                    .theme--light.v-label{
                        color: #ffffff;
                        font-size: 14px;
                    }

                    color: #ffffff;
                }
                .v-input::placeholder{
                    color: #585858 !important;
                }
                .v-input.search{
                    input{
                        color: #ffffff;
                    }
                    .theme--light.v-icon{
                        color: #585858;
                        input{
                         color: #ffffff;
                        }
                    }
                    .v-input__slot{
                        border: 1px solid #585858;
                        padding:6px;
                        background: transparent;
                        border-radius: 20px;
                        .v-text-field__slot{
                            margin-left: 15px;
                        }
                    }
                    .v-input__slot:before{
                        border-color: transparent !important ;

                    }
                    .theme--light.v-label{
                        color: #ffffff;
                        font-size: 14px;
                    }
                }
                .theme--light.v-text-field > .v-input__control > .v-input__slot:before{
                    border-color: #ffffff
                }
                .theme--light.v-icon{
                    color: #ffffff;
                }
            }
        }

        .slider-header{
            @include row;
            align-items: center;
            .title{
            display:flex;
            font-size:26px;
            font-weight: bold;
            font-family:$header-fontfamily;
            margin-right:1em;
            }
            &:after{
                content: '';
                display:inline-block;
                height:1px;
                flex:1;
                background-color:#585858;
            }
        }

        a{
            &.button{
            text-decoration: none;
            filter:none;
            color:white;
            background: $wallet-button-gradient;
            }
            &.center{
            display:block;
            margin: 0 auto;
            }
        }
        a.button{
            width: 230px;
        }

    }
    @media only screen and (max-width: 1199px) {
      .rewards{
        .header{
          .title{
            font-size: 25px;
            margin-bottom: 30px;
          }
          display: flex;
          flex-direction: column;
          .selections{
              display: flex;
              flex-direction: column-reverse;
              .v-input{
                  margin-left: 0px;
              }
          }
        }
      }
    }
  }
</style>
