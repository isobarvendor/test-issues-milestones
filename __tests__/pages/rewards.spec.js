import { mount,  createLocalVue } from '@vue/test-utils';
import Vuex from 'vuex';
import VueRouter from 'vue-router'

import rewards from '@/pages/rewards';

import RewardsCatalogue from '@/components/RewardsCatalogue'


import {mockCMSConfig,mockCMSContent} from "@/__mock__/mock";


import actions from '@/store/actions';
import mutations from '@/store/mutations';


let localVue, store,router;


  beforeEach(() => {
    localVue = createLocalVue();
    localVue.use(Vuex);
    localVue.use(VueRouter);
    router = new VueRouter();
    store = new Vuex.Store({
      getters: {
        getCMSContent() {
          return mockCMSContent;
        },
        getLoginAccount(){
          return {
            name:"ron",
            email:"ron@dsw.com",
            imageUrl:"tes"
          }
        },
        getCMSConfig(){
          return mockCMSConfig[0];
        },
        getListPrize(){
         return  [{
            name:"Reward1",
            prizeId:"prizeid",
            prizeCost:[{ammount :100}],
          },
          {
            name:"Reward2",
            prizeId:"prizeid",
            prizeCost:[{ammount :100}],
          }
        ]
        }
      },
      state:{
        token:"",
        language: "en",
        listWallet:{
          walletStatus:[
            {amount :30}
          ]
        },

      },
      actions,
      mutations,
    });
  });

  const factory = () => {
    return mount(rewards, {
      data(){
        return {
          listPrizes:store.getters.getListPrize
        }
      },
      localVue, store,router
    });
  };



describe('rewards', () => {


  test('renders a RewardsCatalogue', () => {
    const wrapper = factory();
    const component= wrapper.findComponent(RewardsCatalogue);
    expect(component.exists()).toBeTruthy();
  });


  test("mounts properly", () => {
    const wrapper = factory();
    expect(wrapper.vm).toBeTruthy();
  });

  test("renders properly", () => {
    const wrapper = factory();
    expect(wrapper.html()).toMatchSnapshot();
  });

});
