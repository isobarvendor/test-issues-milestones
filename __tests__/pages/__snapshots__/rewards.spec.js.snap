// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rewards renders properly 1`] = `
<div id="main" class="rewards">
  <div class="wrapper">
    <div class="container">
      <div class="header"><span class="title">Rewards</span>
        <div class="selections"></div>
      </div>
      <div class="rewards-catalogue">
        <div class="wallet-desktop"></div>
      </div>
    </div>
  </div>
</div>
`;
